backend/backendutil/search.go

```diff
@@ -194,11 +216,12 @@ func matchSeqNumAndUid(seqNum uint32, uid uint32, c *imap.SearchCriteria) bool {
 }
 
 func matchDate(date time.Time, c *imap.SearchCriteria) bool {
-	date = date.Round(24 * time.Hour)
+	date = time.Date(date.Year(), date.Month(), date.Day(), 0, 0, 0, 0, time.UTC)
```

Comment:
The date isn't guaranteed to be UTC, is it?

Reply:
Time zone here should be the same as in criteria and actual value is not important - comparison is required to be zone unaware.

Reply:
Ah, indeed. Thanks for adding a comment explaining that.